### PR TITLE
Packaging dependencies for wayland in qtile

### PR DESCRIPTION
**Pull request by brunothedev**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

# Introduction

Before getting into the pull request, i would like to tell that this is my first contribution to void-linux/void-packages, so i am sorry if i made any mistakes(I probably added too much dependencies), i would also like to point out that this pull request may be needed to be splitted, since it adds 3 packages and modify 1 package.

## Summary

This pull request add the following packages:
- python-pywayland
- python-xkbcommon
- pywlroots
This pull request modify:
- qtile

## Reason

The reason i made this pull request is to add all the dependencies to use qtile with the wayland backend

### CODE REVIEW

File: srcpkgs/python-pywayland/template [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'python-pywayland'
+pkgname=python-pywayland
```
Comment (classabbyamp) [inline srcpkgs/python-pywayland/template:2]:
```suggestion
pkgname=python3-pywayland
```

File: srcpkgs/python-xkbcommon/template [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Template file for 'python-xkbcommon'
+pkgname=python-xkbcommon
+version=0.4
+revision=1
+makedepends="python3-devel python3-cffi libxkbcommon-devel python3-setuptools"
+depends="python3 python3-cffi libxkbcommon"
```
Comment (classabbyamp) [inline srcpkgs/python-xkbcommon/template:6]:
```suggestion
depends="python3-cffi"
```
python3 is implied by cffi, libxkbcommon should be detected via linking

File: srcpkgs/python-pywayland/template [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+version=0.4.14
+revision=1
+hostmakedepends="pkg-config"
+makedepends="python3-devel python3-cffi wayland-devel python3-setuptools"
```
Comment (classabbyamp) [inline srcpkgs/python-pywayland/template:6]: setuptools should be hostmakedepends

File: srcpkgs/pywlroots/template [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Template file for 'pywlroots'
+pkgname=pywlroots
+version=0.15.24
+revision=1
+makedepends="wlroots-devel python3-devel python3-cffi python-pywayland python-xkbcommon python3-setuptools"
+depends="wlroots python3 python3-cffi python-pywayland python-xkbcommon"
```
Comment (classabbyamp) [inline srcpkgs/pywlroots/template:6]:
```suggestion
depends="wlroots python3-cffi python-pywayland python-xkbcommon"
```

File: srcpkgs/qtile/template [hunk 5]
```diff
@@ -1,11 +1,11 @@
 # Template file for 'qtile'
 pkgname=qtile
 version=0.22.0
-revision=2
+revision=3
 build_style=python3-pep517
 hostmakedepends="python3-setuptools_scm python3-cairocffi python3-xcffib python3-wheel pkg-config"
 makedepends="python3-devel libffi-devel pulseaudio-devel"
-depends="python3-cairocffi python3-xcffib pango gdk-pixbuf"
+depends="python3-cairocffi python3-xcffib pango gdk-pixbuf pywlroots"
```
Comment (classabbyamp) [inline srcpkgs/qtile/template:8]: i don't think this should be specified here, as not everyone using qtile is using the wayland version. Would be better to have a [README.voidlinux](https://github.com/void-linux/void-packages/blob/master/Manual.md#void-specific-documentation) that says to install it for wayland support

Comment (brunothedev) [reply srcpkgs/qtile/template:8]: Honestly i think it should be there, the gnome package for example also installs all wayland dependencies

Comment (ahesford) [reply srcpkgs/qtile/template:8]: We don't force optional dependencies in this way. If you really want this, add a `qtile-wayland` metapackage for convenience.

Comment (brunothedev) [reply srcpkgs/qtile/template:8]: > We don't force optional dependencies in this way. If you really want this, add a `qtile-wayland` metapackage for convenience.

I have written a commit that alert the user of the pywlroots package in the INSTALL.msg

Comment (ahesford) [reply srcpkgs/qtile/template:8]: Don't use `INSTALL.msg` in this manner; use `README.voidlinux` as abby recommended. That's the idiomatic way to document information specifically related to our packaging of the software. `INSTALL.msg` is meant as a last-ditch attempt to convey critical information, mostly about potentially breaking changes, even though these messages show up in an already noisy environment.

**Comment by paper42** [comment]
squash the commits, we want one commit per package

File: srcpkgs/python3-pywayland/template [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# Template file for 'python3-pywayland'
+pkgname=python3-pywayland
+version=0.4.14
+revision=3
```
Comment (paper42) [inline srcpkgs/python3-pywayland/template:4]: start at 1

File: srcpkgs/python3-pywayland/template [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+configure() {
+	python3 -m pywayland.scanner
+	python3 pywayland/ffi_build.py
+}
+
+install() {
+	mkdir -p ${PKGDESTDIR}/usr/lib
+	python3 setup.py install
+}
```
Comment (paper42) [inline srcpkgs/python3-pywayland/template:24]: These functions don't have the right name, so they won't ever run

File: srcpkgs/python3-pywlroots/template [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# Template file for 'python3-pywlroots'
+pkgname=python3-pywlroots
+version=0.15.24
+revision=3
+hostmakedepends="python3-setuptools"
+makedepends="wlroots-devel python3-devel python3-cffi python3-pywayland python3-xkbcommon"
+depends="wlroots python3-cffi python3-pywayland python3-xkbcommon"
+build_style=python3-module
```
Comment (paper42) [inline srcpkgs/python3-pywlroots/template:8]: run the packages through xlint and fix the warnings

**Comment by brunothedev** [comment]
>

**Comment by paper42** [comment]
Please don't open a new PR, instead just update this one. We don't want to lose the review history and unnecessarily spam people.

**Comment by brunothedev** [comment]
> Please don't open a new PR, instead just update this one. We don't want to lose the review history and unnecessarily spam people.

Closing this pull request was a mistake, i am sorry

File: srcpkgs/qtile/template (hunk 5 above)
Comment (sgn) [reply srcpkgs/qtile/template]: or a subpkg for `qtile-wayland` which is a meta package that pull them in

**Comment by ahesford** [comment]
Why are you adding pip to these packages? If a build or test is failing without it, that means you are missing some dependency and the setup script is trying to satisfy it automatically. We don't want pip fetching packages, so you'll need to find out what's missing and add the right dependencies. If the failures are seen only when running tests, those dependencies should be added to `checkdepends`.

**Comment by brunothedev** [comment]
About commit c68794c, i cannot reproduce the tests, even weirder is that i added the 'wheel' dependency, so i ask, is there a way to run the test on machine?

**Comment by classabbyamp** [comment]
use `xbps-src pkg -Q` to run checks

**Comment by brunothedev** [comment]
I already resolved the test errors from python3-pywlroots and python3-pywayland, but python3-xkbcommon throws some errors at testing, i already made a issue about this on: https://github.com/sde1000/python-xkbcommon/issues/17

**Comment by brunothedev** [comment]
I recently migrated from void linux(now using openSuse), so i am closing this PR, but honestly, even before i migrated, i just dind't                      want to maintain all these packages, i had a LOT of frustation with qtile(Configuring is a pain), and me leaving void is the final straw, i am closing this PR.

**Pull request closed without merging**